cumulus/parachains/runtimes/assets/asset-hub-rococo/src/xcm_config.rs

```diff
@@ -0,0 +1,972 @@
+/// ready for dispatching a transaction with Xcm's `Transact`. There is an `OriginKind` which can
+/// biases the kind of local `Origin` it will become.
+pub type XcmOriginToTransactDispatchOrigin = (
+	// Sovereign account converter; this attempts to derive an `AccountId` from the origin location
+	// using `LocationToAccountId` and then turn that into the usual `Signed` origin. Useful for
+	// foreign chains who want to have a local sovereign account on this chain which they control.
+	SovereignSignedViaLocation<LocationToAccountId, RuntimeOrigin>,
+	// Native converter for Relay-chain (Parent) location; will convert to a `Relay` origin when
+	// recognised.
+	RelayChainAsNative<RelayChainOrigin, RuntimeOrigin>,
+	// Native converter for sibling Parachains; will convert to a `SiblingPara` origin when
+	// recognised.
+	SiblingParachainAsNative<cumulus_pallet_xcm::Origin, RuntimeOrigin>,
+	// Superuser converter for the Relay-chain (Parent) location. This will allow it to issue a
+	// transaction from the Root origin.
+	ParentAsSuperuser<RuntimeOrigin>,
+	// Native signed account converter; this just converts an `AccountId32` origin into a normal
+	// `RuntimeOrigin::Signed` origin of the same 32-byte value.
+	SignedAccountId32AsNative<RelayNetwork, RuntimeOrigin>,
+	// Xcm origins can be represented natively under the Xcm pallet's Xcm origin.
+	XcmPassthrough<RuntimeOrigin>,
+);
+
+parameter_types! {
+	pub const MaxInstructions: u32 = 100;
+	pub const MaxAssetsIntoHolding: u32 = 64;
+	pub XcmAssetFeesReceiver: Option<AccountId> = Authorship::author();
+}
+
+match_types! {
+	pub type ParentOrParentsPlurality: impl Contains<MultiLocation> = {
+		MultiLocation { parents: 1, interior: Here } |
+		MultiLocation { parents: 1, interior: X1(Plurality { .. }) }
+	};
+	pub type ParentOrSiblings: impl Contains<MultiLocation> = {
+		MultiLocation { parents: 1, interior: Here } |
+		MultiLocation { parents: 1, interior: X1(_) }
+	};
+}
+
+/// A call filter for the XCM Transact instruction. This is a temporary measure until we properly
+/// account for proof size weights.
+///
+/// Calls that are allowed through this filter must:
+/// 1. Have a fixed weight;
+/// 2. Cannot lead to another call being made;
+/// 3. Have a defined proof size weight, e.g. no unbounded vecs in call parameters.
+pub struct SafeCallFilter;
+impl Contains<RuntimeCall> for SafeCallFilter {
+	fn contains(call: &RuntimeCall) -> bool {
+		#[cfg(feature = "runtime-benchmarks")]
+		{
+			if matches!(call, RuntimeCall::System(frame_system::Call::remark_with_event { .. })) {
+				return true
+			}
+		}
+
+		// Allow to change dedicated storage items (called by governance-like)
+		match call {
+			RuntimeCall::System(frame_system::Call::set_storage { items })
+				if items.iter().all(|(k, _)| k.eq(&bridging::XcmBridgeHubRouterByteFee::key())) ||
+					items.iter().all(|(k, _)| k.eq(&Flavor::key())) =>
+				return true,
+			_ => (),
+		};
```

Comment:
IIUC you set `Flavor` to use `storage` and you plan to do Wococo genesis with the default runtime and then set storage to change its parent?

Reply:
> set storage to change its parent

I am not sure if I under stand this. What parent do you mean?
 `Flavor` is just one storage key/enum-value item which allows us to do some tweaks and adjustments in code based on this storage item value.

Well, we dont need and want another dedicated AssetHubWococo runtime which will be just exact copy of this AssetHubRococo runtime files. With this `Flavor` feature we can just adjust several attributes which are important for bridging, e.g. `UniversalLocation` is important and it should contains correct `GlobalConsensus`
```
pub UniversalLocation: InteriorMultiLocation =
		X2(GlobalConsensus(RelayNetwork::get()), Parachain(ParachainInfo::parachain_id().into()));


impl Get<NetworkId> for RelayNetwork {
	fn get() -> NetworkId {
		match Flavor::get() {
			RuntimeFlavor::Rococo => NetworkId::Rococo,
			RuntimeFlavor::Wococo => NetworkId::Wococo,
		}
	}
}
```

So the result is that:
- by default deployed `asset-hub-rococo-runtime` has `NetworkId::Rococo`
- we also deploy the same `asset-hub-rococo-runtime` as `AssetHubWococo` to the `Wococo` world, so with `Flavor::Wococo` in storage, we achieve that it returns `NetworkId::Wococo`


Yes, it is two step process for AssetHubWococo:
- deploy runtime
- trigger governance-like call `Call::set_storage(Flavor::key(), Flavor::Wococo)`


I dont know if there is easier and nicer way, but at the begining I was not sure how many tweaks will be needed,
Maybe just to store `NetworkId` or `UniversalLocation` to the storage directly (it is xcm versioned stuff, so maybe just another problems with migration to higher version).

Reply:
I see, OK I was basically wondering why `pub storage` and not `pub const`, and I suppose the idea is that it defaults to Rococo, but either with genesis storage or with a `set_storage` you can set this to Wococo.

Reply:
>  and I suppose the idea is that it defaults to Rococo, but either with genesis storage or with a `set_storage` you can set this to Wococo.

yes, exactly :)